bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```diff
@@ -446,13 +446,13 @@ public void run() {
             executor.shutdown();
             try {
                 if (!executor.awaitTermination(environment.getTimeout(), TimeUnit.SECONDS)) {
-                    // For some reason we've timed out. The deletion should likely be executing, but let's force it to
-                    // be safe.
-                    cleaner.cleanup();
+                    // For some reason we've timed out. The deletion should likely be executing.
+                    // We can't start a new cleanup to force it. On Windows we would have the side effect to have 2 cleaner processes to
+                    // be executed, with the risk that a new installation has been installed and the new cleaner cleaning the new installation
+                    log.cleanupTimeout(environment.getTimeout(), environment.getJBossHome());
```

Comment:
Maybe we should just execute the `cleaner.cleanup()` in a try/catch where we simply ignore the IOException and, like you do here, log the exception.

Reply:
If we do call cleaner.cleanup, even in a try/catch, it will fork a new process and leave. With very special scheduling, in which the installation dir has been already populated with some content (because the initial cleaner actually cleaned everything), the new process will delete the new installation. I suspect that the random failure is caused by this scenario,